### ISSUE TITLE
UI feedback for crawl overruns

### ISSUE THREAD

**Issue by ronaldknol** [shirosaidev/diskover#4]
I'm indexing a volume with 47,900 directories and 267,067 files.
The output is like this:

```
[2017-05-01 08:14:20] [status] Finding directories to crawl
Crawling: [100%] |████████████████████████████████████████| 4270/4270
Crawling: [100%] |████████████████████████████████████████| 7674/7674
Crawling: [100%] |████████████████████████████████████████| 7767/7767
Crawling: [100%] |████████████████████████████████████████| 9119/9119
Crawling: [100%] |████████████████████████████████████████| 9211/9211
Crawling: [100%] |████████████████████████████████████████| 9476/9476
Crawling: [100%] |████████████████████████████████████████| 10038/10038
Crawling: [100%] |████████████████████████████████████████| 10239/10239
Crawling: [100%] |████████████████████████████████████████| 10392/10392
Crawling: [100%] |████████████████████████████████████████| 10668/10668
Crawling: [100%] |████████████████████████████████████████| 10761/10761
Crawling: [100%] |████████████████████████████████████████| 10761/10761
Crawling: [100%] |████████████████████████████████████████| 10925/10925
Crawling: [100%] |████████████████████████████████████████| 10925/10925
Crawling: [100%] |████████████████████████████████████████| 11008/11008
Crawling: [100%] |████████████████████████████████████████| 11064/11064
Crawling: [100%] |████████████████████████████████████████| 11130/11130
Crawling: [100%] |████████████████████████████████████████| 11213/11213
Crawling: [100%] |████████████████████████████████████████| 11289/11289
Crawling: [100%] |████████████████████████████████████████| 11433/11433
Crawling: [100%] |████████████████████████████████████████| 11513/11513
Crawling: [100%] |████████████████████████████████████████| 16255/16255
Crawling: [100%] |████████████████████████████████████████| 17602/17602
Crawling: [100%] |████████████████████████████████████████| 18254/18254
Crawling: [100%] |████████████████████████████████████████| 18254/18254
Crawling: [100%] |████████████████████████████████████████| 18544/18544
Crawling: [100%] |████████████████████████████████████████| 18544/18544
Crawling: [100%] |████████████████████████████████████████| 18679/18679
Crawling: [100%] |████████████████████████████████████████| 19131/19131
Crawling: [100%] |████████████████████████████████████████| 19131/19131
Crawling: [100%] |████████████████████████████████████████| 19282/19282
Crawling: [100%] |████████████████████████████████████████| 19282/19282
Crawling: [100%] |████████████████████████████████████████| 19628/19628
Crawling: [100%] |████████████████████████████████████████| 19628/19628
Crawling: [100%] |████████████████████████████████████████| 19926/19926
Crawling: [100%] |████████████████████████████████████████| 21296/21296
Crawling: [100%] |████████████████████████████████████████| 21296/21296
Crawling: [100%] |████████████████████████████████████████| 21623/21623
Crawling: [100%] |████████████████████████████████████████| 21623/21623
Crawling: [100%] |████████████████████████████████████████| 22093/22093
Crawling: [100%] |████████████████████████████████████████| 22196/22196
Crawling: [100%] |████████████████████████████████████████| 22196/22196
Crawling: [100%] |████████████████████████████████████████| 22480/22480
Crawling: [100%] |████████████████████████████████████████| 22649/22649
Crawling: [100%] |████████████████████████████████████████| 22649/22649
Crawling: [100%] |████████████████████████████████████████| 22819/22819
Crawling: [100%] |████████████████████████████████████████| 22819/22819
Crawling: [100%] |████████████████████████████████████████| 23464/23464
Crawling: [100%] |████████████████████████████████████████| 23591/23591
Crawling: [100%] |████████████████████████████████████████| 23716/23716
Crawling: [100%] |████████████████████████████████████████| 23841/23841
Crawling: [100%] |████████████████████████████████████████| 23965/23965
Crawling: [100%] |████████████████████████████████████████| 23965/23965
Crawling: [100%] |████████████████████████████████████████| 24245/24245
Crawling: [100%] |████████████████████████████████████████| 24245/24245
Crawling: [100%] |████████████████████████████████████████| 24513/24513
Crawling: [100%] |████████████████████████████████████████| 25235/25235
Crawling: [100%] |████████████████████████████████████████| 25366/25366
Crawling: [100%] |████████████████████████████████████████| 25488/25488
Crawling: [100%] |████████████████████████████████████████| 25612/25612
Crawling: [100%] |████████████████████████████████████████| 25815/25815
Crawling: [100%] |████████████████████████████████████████| 25888/25888
Crawling: [100%] |████████████████████████████████████████| 25888/25888
Crawling: [100%] |████████████████████████████████████████| 25971/25971
Crawling: [100%] |████████████████████████████████████████| 25971/25971
Crawling: [100%] |████████████████████████████████████████| 26105/26105
Crawling: [100%] |████████████████████████████████████████| 26178/26178
Crawling: [100%] |████████████████████████████████████████| 26607/26607
Crawling: [100%] |████████████████████████████████████████| 26690/26690
Crawling: [100%] |████████████████████████████████████████| 26869/26869
Crawling: [100%] |████████████████████████████████████████| 26955/26955
Crawling: [100%] |████████████████████████████████████████| 26955/26955
Crawling: [100%] |████████████████████████████████████████| 27043/27043
Crawling: [100%] |████████████████████████████████████████| 27043/27043
Crawling: [100%] |████████████████████████████████████████| 27268/27268
Crawling: [100%] |████████████████████████████████████████| 27391/27391
Crawling: [100%] |████████████████████████████████████████| 27458/27458
Crawling: [100%] |████████████████████████████████████████| 27458/27458
Crawling: [100%] |████████████████████████████████████████| 27541/27541
Crawling: [100%] |████████████████████████████████████████| 27698/27698
Crawling: [100%] |████████████████████████████████████████| 27904/27904
Crawling: [100%] |████████████████████████████████████████| 27904/27904
Crawling: [100%] |████████████████████████████████████████| 28012/28012
Crawling: [100%] |████████████████████████████████████████| 28781/28781
Crawling: [100%] |████████████████████████████████████████| 28902/28902
Crawling: [100%] |████████████████████████████████████████| 28902/28902
Crawling: [100%] |████████████████████████████████████████| 29472/29472
Crawling: [100%] |████████████████████████████████████████| 29472/29472
Crawling: [100%] |████████████████████████████████████████| 29575/29575
Crawling: [100%] |████████████████████████████████████████| 29575/29575
Crawling: [100%] |████████████████████████████████████████| 30642/30642
Crawling: [100%] |████████████████████████████████████████| 30919/30919
Crawling: [100%] |████████████████████████████████████████| 31302/31302
Crawling: [100%] |████████████████████████████████████████| 31396/31396
Crawling: [100%] |████████████████████████████████████████| 32447/32447
Crawling: [100%] |████████████████████████████████████████| 32714/32714
Crawling: [100%] |████████████████████████████████████████| 33745/33745
Crawling: [100%] |████████████████████████████████████████| 34355/34355
Crawling: [100%] |████████████████████████████████████████| 34355/34355
Crawling: [100%] |████████████████████████████████████████| 34647/34647
Crawling: [100%] |████████████████████████████████████████| 34647/34647
Crawling: [100%] |████████████████████████████████████████| 35524/35524
Crawling: [100%] |████████████████████████████████████████| 35524/35524
Crawling: [100%] |████████████████████████████████████████| 35824/35824
Crawling: [100%] |████████████████████████████████████████| 36792/36792
Crawling: [100%] |████████████████████████████████████████| 36929/36929
Crawling: [100%] |████████████████████████████████████████| 37980/37980
Crawling: [100%] |████████████████████████████████████████| 38666/38666
Crawling: [100%] |████████████████████████████████████████| 38666/38666
Crawling: [100%] |████████████████████████████████████████| 38814/38814
Crawling: [100%] |████████████████████████████████████████| 39529/39529
Crawling: [100%] |████████████████████████████████████████| 39674/39674
Crawling: [100%] |████████████████████████████████████████| 40188/40188
Crawling: [100%] |████████████████████████████████████████| 40315/40315
Crawling: [100%] |████████████████████████████████████████| 41654/41654
Crawling: [100%] |████████████████████████████████████████| 41854/41854
Crawling: [100%] |████████████████████████████████████████| 42437/42437
Crawling: [100%] |████████████████████████████████████████| 44910/44910
Crawling: [100%] |████████████████████████████████████████| 44910/44910
Crawling: [100%] |████████████████████████████████████████| 45104/45104
Crawling: [100%] |████████████████████████████████████████| 45219/45219
Crawling: [100%] |████████████████████████████████████████| 45426/45426
Crawling: [100%] |████████████████████████████████████████| 45539/45539
Crawling: [100%] |████████████████████████████████████████| 45643/45643
Crawling: [100%] |████████████████████████████████████████| 45643/45643
Crawling: [100%] |████████████████████████████████████████| 45753/45753
Crawling: [100%] |████████████████████████████████████████| 45850/45850
Crawling: [100%] |████████████████████████████████████████| 45850/45850
Crawling: [100%] |████████████████████████████████████████| 46298/46298
Crawling: [100%] |████████████████████████████████████████| 46298/46298
Crawling: [100%] |████████████████████████████████████████| 46421/46421
Crawling: [100%] |████████████████████████████████████████| 46657/46657
Crawling: [100%] |████████████████████████████████████████| 46657/46657
Crawling: [100%] |████████████████████████████████████████| 46850/46850
Crawling: [100%] |████████████████████████████████████████| 46954/46954
Crawling: [100%] |████████████████████████████████████████| 47045/47045
Crawling: [100%] |████████████████████████████████████████| 47136/47136
Crawling: [100%] |████████████████████████████████████████| 47227/47227
Crawling: [100%] |████████████████████████████████████████| 47323/47323
Crawling: [100%] |████████████████████████████████████████| 47419/47419
Crawling: [100%] |████████████████████████████████████████| 47522/47522
Crawling: [100%] |████████████████████████████████████████| 47625/47625
Crawling: [100%] |████████████████████████████████████████| 47736/47736
Crawling: [100%] |████████████████████████████████████████| 47736/47736
Crawling: [100%] |████████████████████████████████████████| 47899/47900
[2017-05-01 08:26:19] [info] Directories Crawled: 47900
```

Somewhere the crawl percentage wraps.

**Comment by shirosaidev** [comment]
Do you always see this or just randomly? Can you reply with the full command lines you are using for the crawl.

**Comment by ronaldknol** [comment]
I ran the default command ```cd /directory_to_index ; python ~/diskover/diskover.py``` which generated that odd crawl output.
I think the code is running through the directory content faster than is gets new directories.
I am now running it with ```python diskover.py -s 0 -m 0 -i diskover.dir.20170501 -d /dir_to_index``` and it seems to not overrun, so it's spending more time on each directory and giving the find time to get more directories.

**Comment by shirosaidev** [comment]
I released version **1.0.4** tonight which should fix this issue. Please update and reply if it's fixed.

**Comment by ronaldknol** [comment]
Hmmm... 1.0.4 fails on my system right away:

```
root@linserver03:~/diskover# python diskover.py -s 0 -m 0 -i diskover.idx -d /mnt
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "diskover.py", line 314, in processDirectoryWorker
    filelist = crawlFiles(path, DATEEPOCH, DAYSOLD, MINSIZE, EXCLUDED_FILES, VERBOSE)
  File "diskover.py", line 270, in crawlFiles
    "owner": "%s" % owner.decode('utf-8'),
AttributeError: 'int' object has no attribute 'decode'

root@linserver03:~/diskover# python -V
Python 2.7.12
```
Running it with Python3 gets me:

```
root@linserver03:~/diskover# python3 diskover.py -s 0 -m 0 -i diskover.idx -d /mnt
Traceback (most recent call last):
  File "diskover.py", line 14, in <module>
    import Queue
ImportError: No module named 'Queue'
```

**Comment by shirosaidev** [comment]
I've released **version 1.0.5** which should fix this. I haven't done any testing using Python 3.

**Comment by shirosaidev** [comment]
This was a bug caused by the crawl going over nfs and not being able to grab owner/group names and falls back to uid/gid numbers. Unicode decoding was failing on integer values, was expecting a string.

**Comment by ronaldknol** [comment]
Indeed fixed with 1.0.5
